luda-editor/luda-editor-client/src/app/storage/storage.rs

```diff
@@ -0,0 +1,109 @@
+use super::{
+    fetch_background_images::FetchBackgroundImagesError,
+    fetch_character_images::FetchCharacterImagesError,
+    get_background_image::GithubStorageBackgroundImageGet,
+    get_background_image_paths::GithubStorageBackgroundImagePathsGet,
+    get_character_image::GithubStorageCharacterImageGet,
+    get_character_image_paths::GithubStorageCharacterImagePathsGet, get_meta::GithubStorageMetaGet,
+    get_sequence::GithubStorageSequenceGet, get_sequence_list::GithubStorageSequenceListGet,
+    get_sequence_lock_state::StorageSequenceLockStateGet,
+    get_sequence_titles::GithubStorageSequenceTitlesGet, lock_sequence::GithubStorageSequenceLock,
+    put_sequence::GithubStorageSequencePut, put_sequence_titles::GithubStorageSequenceTitlesPut,
+    unlock_sequence::GithubStorageSequenceUnlock,
+};
+use crate::app::github_api::GithubAPiClient;
+use async_trait::async_trait;
+use dashmap::DashSet;
+use futures::join;
+use namui::prelude::*;
+use std::{fmt::Debug, sync::Arc};
+
+#[async_trait(?Send)]
+pub trait GithubStorage:
+    Debug
+    + Send
+    + Sync
+    + GithubStorageMetaGet
+    + StorageSequenceLockStateGet
+    + GithubStorageSequenceLock
+    + GithubStorageSequenceUnlock
+    + GithubStorageSequenceGet
+    + GithubStorageSequencePut
+    + GithubStorageSequenceListGet
+    + GithubStorageSequenceTitlesGet
+    + GithubStorageSequenceTitlesPut
+    + GithubStorageBackgroundImagePathsGet
+    + GithubStorageBackgroundImageGet
+    + GithubStorageCharacterImageGet
+    + GithubStorageCharacterImagePathsGet
```

Comment:
Suuuuper long trait.
This is to divde the impls into several files while using one trait for easy mocking.

below is impossible for now.
``` rust
trait Trait {
  fn a();
  fn b();
}
struct Struct;

// a.rs
impl Trait for Struct {
  fn a() {}
}

// b.rs
impl Trait for Struct {
  fn b() {}
}
```

Reply:
`StorageSequenceLockStateGet` name fixed.

Reply:
Internal traits (GithubStorageMetaGet, ...) are not exposed outside GithubStorage.